### PR TITLE
docs: Add Druid 28 release notes and upgrade notes

### PR DESCRIPTION
**Pull request by ektravel**
This PR replaces https://github.com/apache/druid/pull/15173.

This PR adds a release notes page for Druid 28.0.1 and an upgrade notes page for Druid releases.

I will open a separate PR with an empty template for Druid 29.0.0 on master.

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

File: docs/release-info/release-notes.md [hunk 1]
```diff
@@ -0,0 +1,882 @@
...
+
+Druid 28.0.0 adds experimental support for concurrent append and replace.
+This feature allows you to safely replace the existing data in an interval of a datasource while new data is being appended to that interval. One of the most common applications of this is appending new data to an interval while compaction of that interval is already in progress.
+For more information, see [Concurrent append and replace](https://druid.apache.org/docs/latest/data-management/automatic-compaction#concurrent-append-and-replace).
+
+Segment locking will be deprecated and removed in favor of concurrent append and replace that is much simpler in design. With concurrent append and replace, Druid doesn't lock compaction jobs out because of active realtime ingestion.
+
+### Task locks for append and replace batch ingestion jobs
+
+Append batch ingestion jobs can now share locks. This allows you to run multiple append batch ingestion jobs against the same time internal. Replace batch ingestion jobs still require an exclusive lock. This means you can run multiple append batch ingestion jobs and one replace batch ingestion job for a given interval.
```
Comment (317brian) [inline docs/release-info/release-notes.md:170]:
```suggestion
Append batch ingestion jobs can now share locks. This allows you to run multiple append batch ingestion jobs against the same time interval. Replace batch ingestion jobs still require an exclusive lock. This means you can run multiple append batch ingestion jobs and one replace batch ingestion job for a given interval.
```

**Review by 317brian — APPROVED** [review]
LGTM, this is the same content we put out with the release on GH

File: docs/release-info/release-notes.md [hunk 2]
```diff
@@ -0,0 +1,882 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+  
+Apache Druid 28.0.0 contains over $NUMBER_FEATURES new features, bug fixes, performance enhancements, documentation improvements, and additional test coverage from $NUMBER_OF_CONTRIBUTORS contributors.
```
Comment (317brian) [inline docs/release-info/release-notes.md:25]:
```suggestion
Apache Druid 28.0.0 contains over 420 new features, bug fixes, performance enhancements, documentation improvements, and additional test coverage from 57 contributors.
```

File: docs/release-info/release-notes.md [hunk 3]
```diff
@@ -28,7 +28,10 @@ Apache Druid 28.0.1 is a patch release that fixes some issues in the Druid 28.0.
 
 ### Notable improvements
```
Comment (LakshSingla) [inline docs/release-info/release-notes.md:29]: These should be bug fixes IMO. Many of these were regressions introduced in this release or something that would have likely become more common due to the changes in this release.

**Review by LakshSingla — APPROVED** [review]

**Pull request merged**